### PR TITLE
📖  Bento: Adopt new formatting for `amp-base-carousel` documentation

### PR DESCRIPTION
**Pull request by caroqliu**
Related to #28284, incorporating format changes from #31944

### CODE REVIEW

File: extensions/amp-base-carousel/amp-base-carousel.md [hunk 1]
```diff
@@ -40,9 +40,204 @@ navigational arrows to go forward or backwards a single item.
...
+```
+
+**goToSlide(index: number)**
+Moves the carousel to the slide specified by the `index` argument.
+
+```
+api.goToSlide(0); // Advance to first slide.
+api.goToSlide(length - 1); // Advance to last slide.
+```
+
```
Comment (CrystalOnScript) [inline extensions/amp-base-carousel/amp-base-carousel.md:140]: Do they need to include `index:` in the argument? Or just the index number?

**Review by CrystalOnScript — APPROVED** [review]
This looks amazing!!! I was mistaken about executable code samples and Bento, so just one comment about updating the code sample and a qq about one of the actions. Otherwise LGTM!

File: extensions/amp-base-carousel/amp-base-carousel.md (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/amp-base-carousel.md:140]: Just the index. The `index: number` is to give the parameter a type, but let me specify this inline instead so it's less confusing, something like this:
```suggestion
**goToSlide(index)**
Args:
  - `index`: number [required]
Moves the carousel to the slide specified by the `index` argument.

\```
api.goToSlide(0); // Advance to first slide.
api.goToSlide(length - 1); // Advance to last slide.
\```

```

Comment (dvoytenko) [reply extensions/amp-base-carousel/amp-base-carousel.md:140]: TypeScript-like notation here I think is the best in the long term. We are also likely to publish TypeScript type declarations (based on our `x.type.js`) in the future too.

File: extensions/amp-base-carousel/amp-base-carousel.md [hunk 2]
```diff
@@ -40,9 +40,204 @@ navigational arrows to go forward or backwards a single item.
...
+```
+<link rel="stylesheet" type="text/css" href="https://cdn.ampproject.org/v0/amp-base-carousel-1.0.css">
+```
+
+Fully valid AMP pages use the AMP layout system to infer sizing of elements to create a page structure before downloading any remote resources. However, Bento use imports components into less controlled environments and AMP's layout system is inaccessible.
+
+**Container type**
+
+The `amp-base-carousel` component has a defined layout size type. To ensure the component renders correctly, be sure to apply a size to the component and its immediate children (slides) via `height` and `width` or `aspect-ratio`:
```
Comment (dvoytenko) [inline extensions/amp-base-carousel/amp-base-carousel.md:166]: Maybe "via a desired CSS layout (such as `height`, `width`, `aspect-ratio`, and other)"? Just want to stress that any CSS layout rules would work.

File: extensions/amp-base-carousel/amp-base-carousel.md [hunk 3]
```diff
@@ -40,9 +40,204 @@ navigational arrows to go forward or backwards a single item.
...
+addition, when changing `visible-slides`, you likely want to change
+`advance-count`.
+
+```html
+<!-- Using an aspect ratio of 3:2 for the slides in this example. -->
+<amp-base-carousel
+  layout="responsive"
+  width="3"
+  height="1"
+  heights="(min-width: 600px) calc(100% * 4 * 3 / 2), calc(100% * 3 * 3 / 2)"
```
Comment (dvoytenko) [inline extensions/amp-base-carousel/amp-base-carousel.md:233]: Hmm. I think we should continue advising to use CSS for this. E.g. 

```
<style>
  #carousel1 {
    height: calc(100% * 3 * 3 / 2);
  }
  @media (min-width: 600px) {
    #carousel1 {
      height: calc(100% * 4 * 3 / 2);
    }
  }
</style>
```

Of course, that'd split the media queries between the element and the attributes. But IMHO it's better to fix one part of it than both have two parts "broken". We can also communicate later that we are pursuing Web spec update to support attribute media queries as well.

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: extensions/amp-base-carousel/amp-base-carousel.md (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/amp-base-carousel.md:140]: @dvoytenko are you suggesting keep as is?

File: extensions/amp-base-carousel/amp-base-carousel.md (hunk 3 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/amp-base-carousel.md:233]: To clarify, this section is moved verbatim from its former location in the file, and the code samples here were created with the intention of being used on valid AMP documents. Do you still recommend making the change? Or perhaps duplicating the section so there is one version for valid AMP documents (status quo) and one version with the suggested CSS for Bento documents?

File: extensions/amp-base-carousel/amp-base-carousel.md (hunk 2 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/amp-base-carousel.md:166]: Done.

File: extensions/amp-base-carousel/amp-base-carousel.md [hunk 4]
```diff
@@ -40,9 +40,204 @@ navigational arrows to go forward or backwards a single item.
...
+<!-- TODO: Remove backticks from link when guide is live -->
+
+Bento AMP allows you to use AMP components in non-AMP pages without needing to commit to fully valid AMP. You can take these components and place them in implementations with frameworks and CMSs that don't support AMP. Read more in our guide `[Use AMP components in non-AMP pages](https://amp.dev/documentation/guides-and-tutorials/start/bento_guide/)`.
+
+#### Example
+
+The example below demonstrates `amp-base-carousel` component in standalone use.
+
+```
+<head>
```
[the remark this answers is not in the document]
Comment (caroqliu) [reply extensions/amp-base-carousel/amp-base-carousel.md:54]: Done.

File: extensions/amp-base-carousel/amp-base-carousel.md (hunk 3 above)
Comment (dvoytenko) [reply extensions/amp-base-carousel/amp-base-carousel.md:236]: If it's for valid AMP, then let's keep as is. Sorry for the confusion. But I think we'll be stumbling into these deltas for some time.

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**